### PR TITLE
Refactor julia 1.6

### PR DESCRIPTION
**Pull request by cstich**
This compiles julia 1.6 from source. So far it skips all the tests and I know those need more work. I based it off the recent refactor of julia 1.0 (https://github.com/NixOS/nixpkgs/pull/117881)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/julia/precompile_succeeded.patch [hunk 1]
```diff
@@ -0,0 +1,156 @@
+{ fetchzip, fetchFromGitHub, lib, pkgs, stdenv
```
Comment (samuela) [inline pkgs/development/compilers/julia/precompile_succeeded.patch:1]: This file looks the same as 1.6.nix. Am I missing something?

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -10717,6 +10717,10 @@ in
     inherit (darwin.apple_sdk.frameworks) CoreServices ApplicationServices;
   };
 
+  julia_16 = callPackage ../development/compilers/julia/1.6.nix {
+    inherit (darwin.apple_sdk.frameworks) CoreServices ApplicationServices;
```
Comment (samuela) [inline pkgs/top-level/all-packages.nix:10721]: It seems like these inputs are no longer needed in the derivation? It looks like they've been commented out so they can prob be removed here as well.

File: pkgs/development/compilers/julia/precompile_succeeded.patch (hunk 1 above)
Comment (cstich) [reply pkgs/development/compilers/julia/precompile_succeeded.patch:1]: I am currently traveling so I didn't have time to compare this with the other pull request. I was thinking maybe bits and bobs of this one might be still useful to someone.

Comment (samuela) [reply pkgs/development/compilers/julia/precompile_succeeded.patch:1]: No rush! I just meant that this doesn't look like a patch file. It looks more like the 1.6.nix file from this PR, but perhaps I'm missing something!

**Comment by NickCao** [comment]
julia_16 has been dropped and this is no longer relevant.

**Pull request closed without merging**